test/extensions/resource_monitors/fixed_heap/fixed_heap_monitor_test.cc

```diff
@@ -39,6 +41,30 @@ class ResourcePressure : public Server::ResourceUpdateCallbacks {
 };
 
 TEST(FixedHeapMonitorTest, ComputesCorrectUsage) {
+
+  TestScopedRuntime scoped_runtime;
+  scoped_runtime.mergeValues({{"envoy.reloadable_features.do_not_count_mapped_pages_as_free", "false"}});
+
+  envoy::extensions::resource_monitors::fixed_heap::v3::FixedHeapConfig config;
+  config.set_max_heap_size_bytes(1000);
+  auto stats_reader = std::make_unique<MockMemoryStatsReader>();
+  EXPECT_CALL(*stats_reader, reservedHeapBytes()).WillOnce(testing::Return(800));
+  EXPECT_CALL(*stats_reader, unmappedHeapBytes()).WillOnce(testing::Return(100));
+  EXPECT_CALL(*stats_reader, freeMappedHeapBytes()).WillOnce(testing::Return(200));
+  std::unique_ptr<FixedHeapMonitor> monitor(new FixedHeapMonitor(config, std::move(stats_reader)));
+
+  ResourcePressure resource;
+  monitor->updateResourceUsage(resource);
+  EXPECT_TRUE(resource.hasPressure());
+  EXPECT_FALSE(resource.hasError());
```

Comment:
think these would be better suited as ASSERT_TRUE/FALSE, otherwise if we don't have pressure and the expectation fails we still fall into `resource.pressure()` which will get a value from an empty optional